### PR TITLE
Fix Ddoc comments

### PR DESCRIPTION
**Pull request by wilzbach**
A couple of Ddoc weirdnesses that I observed while playing with https://github.com/dlang/phobos/pull/5495 and the blacklist generator.

### CODE REVIEW

File: std/datetime/timezone.d [hunk 1]
```diff
@@ -3347,7 +3347,7 @@ else version(Posix)
     does.
 
     Params:
-        windowsZonesXMLFileText The text from
+        windowsZonesXMLText = The text from
```
Comment (wilzbach) [inline std/datetime/timezone.d]: The Dscanner check on mismatched Ddoc parameters didn't recognize this line before ...

File: std/regex/package.d [hunk 2]
```diff
@@ -350,7 +350,7 @@ public alias StaticRegex(Char) = std.regex.internal.ir.StaticRegex!(Char);
     the same character width as $(D pattern).
 
     Params:
-    pattern(s) = Regular expression(s) to match
+    patterns = Regular expression(s) to match
```
Comment (CyberShadow) [inline std/regex/package.d]: Can we put both `pattern` and `patterns` here? If so, how about:

```
pattern = A single regular expression to match.
patterns = An array of regular expression strings. The resulting `Regex` object will match any expression; use $(LREF whichPattern) to know which.
```

**Comment by wilzbach** [comment]
> Can we put both `pattern` and `patterns` here? If so, how about:

Yep I think so:

image

Btw I just saw that with Ddoc `LREF` doesn't always work, for example on the booktable overview at [std.regex](https://dlang.org/phobos/std_regex.html) for `regex` the following is generated with LREF:

https://dlang.org/phobos/std_regex.html#regex (generated by LREF)

However, only an id element with `.` seem to be there:

https://dlang.org/phobos/std_regex.html#.regex

For all links with only a single overload (no dittos) it works.
The issue is with the generated HTML:

```html
<div><span class="def-anchor" id=".regex"></span>
    <div class="quickindex" id="quickindex.regex"></div>@trusted auto <span class="def-anchor" id="regex"></span><code class="ddoc_psymbol">regex</code><span class="template_param_list" title="Template parameter list">(S)</span>(S[] <code class="ddoc_param">patterns</code>, const(char)[] <code class="ddoc_param">flags</code> = "")
    <br><span class="constraint">if (isSomeString!S)</span>;
    <br>
    <br>@trusted auto <span class="def-anchor" id="regex"></span><code class="ddoc_psymbol">regex</code><span class="template_param_list" title="Template parameter list">(S)</span>(S <code class="ddoc_param">pattern</code>, const(char)[] <code class="ddoc_param">flags</code> = "")
    <br><span class="constraint">if (isSomeString!S)</span>;
    <br>
</div>
```

`id=regex` is no longer unique ...
We basically have two paths to go
1) Change LREF to always use `.` (like MYREF) - Ddox creates proper URLs anyways
2) Fix Ddoc's output at DMD 

### LREF

```
std_consolidated.ddoc:58:LREF = <a href="#$1">$(D $1)</a>
std.ddoc:14:LREF = <a href="#.$1">$(D $1)</a>
std-ddox.ddoc:2:LREF = $(D $1)
```

### MYREF

Q: What was the reason for this macro? Can we get rid of it (i.e. use `LREF` everywhere)?

```
std.ddoc:27:MYREF = <a href="#.$1">$(TT $1)</a>$(NBSP)
std-ddox-override.ddoc:1:MYREF = $(D $1)
```

### Ddoc anchor symbols

```
html.ddoc
7:ADEF = <span class="def-anchor" id="$0"></span>
163:DDOC_ANCHOR  = $(ADEF $1)

dlang.org.ddoc
129:DDOC_PSYMBOL = $(ADEF $0)$(TC code, ddoc_psymbol, $0)
130:DDOC_ANCHOR = $(ADEF .$1)$(DIVCID quickindex, quickindex.$1, )
```

I guess changing LREF to behave like MYREF and removing MYREf in the process is the easiest way?

**Comment by CyberShadow** [comment]
> Change LREF to always use . (like MYREF) - Ddox creates proper URLs anyways

That would break existing links to fields/methods.

I think we need a macro for structured links to things on the same page, like XREF. For DDox it would expand to the fully qualified symbol name; for DDoc, the `#.type.fieldOrMethod` link.

**Pull request merged**